iquip/apps/explorer.py

```diff
@@ -31,6 +30,45 @@ def __init__(self, parent: Optional[QWidget] = None):
         layout.addWidget(self.openButton)
 
 
+class FileFinderThread(QThread):
+    """QThread for finding the file list using a command line.
+
+    Signals:
+        finished(experimentList, parent): Fetching the file list is finished.
+
+    Attributes:
+        path: The path of the directory to search experiment files.
+        parent: The widget corresponding to the path.
+    """
+
+    finished = pyqtSignal(list, object)
+
+    def __init__(
+        self,
+        path: str,
+        parent: Union[QTreeWidget, QTreeWidgetItem],
+        callback: Callable[[List[str], Union[QTreeWidget, QTreeWidgetItem]], None]
```

Comment:
Oh, if it is only for the internal use, it would be okay. However, since you named it as a public class, I think it should be considered as a part of the public API.
For the public API, please make it general as possible!

Reply:
I intended to use the thread class only internally, so I will make it private.